WeakAurasOptions/WeakAurasOptions.lua

```diff
@@ -1726,6 +1726,148 @@ function OptionsPrivate.OpenTriggerTemplate(data, targetId)
   end
 end
 
+OptionsPrivate.currentDynamicTextInput = false;
+
+local BaseDynamicTextCodes = {
+  {type = "mini", triggerNum = 0, name = "p", o = 1, n = 7, desc = L["Progress - The remaining time of a timer, or a non-timer value"]},
+  {type = "mini", triggerNum = 0, name = "t", o = 2, n = 7, desc = L["Total - The maximum duration of a timer, or a maximum non-timer value"]},
+  {type = "mini", triggerNum = 0, name = "n", o = 3, n = 7, desc = L["Name - The name of the display (usually an aura name), or the display's ID if there is no dynamic name"]},
+  {type = "mini", triggerNum = 0, name = "i", o = 4, n = 7, desc = L["Icon - The icon associated with the display"]},
+  {type = "mini", triggerNum = 0, name = "s", o = 5, n = 7, desc = L["Stacks - The number of stacks of an aura (usually)"]},
+  {type = "mini", triggerNum = 0, name = "c", o = 6, n = 7, desc = L["Custom - Allows you to define a custom Lua function that returns a list of string values. %c1 will be replaced by the first value returned, %c2 by the second, etc."]},
+  {type = "mini", triggerNum = 0, name = "%", o = 7, n = 7, desc = L["% - To show a percent sign"]},
+}
+
+function OptionsPrivate.UpdateTextReplacements(frame, data)
+  frame.scrollList:ReleaseChildren()
+
+  local props = OptionsPrivate.Private.GetAdditionalProperties(data)
+  local sortedProps = {
+    {type = "header", triggerNum = 0, name = "Global Properties"}
+  }
+  for i = 1, 8 do
+    tinsert(sortedProps, {type = "marker", triggerNum = 0, name = "{rt"..i.."}", o = i, n = 8, desc = "|TInterface\\TargetingFrame\\UI-RaidTargetingIcon_"..i..":0|t"})
+  end
+  tAppendAll(sortedProps, BaseDynamicTextCodes)
+  for triggerNum, triggerProps in pairs(props) do
+    if next(props[triggerNum]) then
+      table.insert(sortedProps, {type = "header", triggerNum = triggerNum, name = OptionsPrivate.GetTriggerTitle(data, triggerNum)})
+      for i = 1, 5 do
+        local prop = CopyTable(BaseDynamicTextCodes[i])
+        prop.triggerNum = triggerNum
+        prop.n = 5
+        table.insert(sortedProps, prop)
+      end
+      for name, desc in pairs(triggerProps) do
```

Comment:
you can simplify the code if you instead of inserting triggerProps directly into sortedPropes, you insert them into a tempPropts, sort tempProps by name and then add them in their sorted order into sortedProps.

This removes the need for the complicated sorter belwo, and for props.o.